htlcswitch/decayedlog.go

```diff
@@ -253,6 +297,191 @@ func (d *DecayedLog) gcExpiredHashes(height uint32) (uint32, error) {
 	return numExpiredHashes, nil
 }
 
+// gcExpiredSets purges the batchReplay bucket of all entries whose CLTV
+// expires below the provided height.
+func (d *DecayedLog) gcExpiredSets(height uint32) (int, error) {
+	breakForEach := errors.New("break out of ForEach")
+
+	numExpiredSets := 0
+	err := kvdb.Batch(d.db, func(tx kvdb.RwTx) error {
+		// There may be replay sets without expiry information.
+		if err := InitReplaySetExpiryForOldSets(tx); err != nil {
+			return err
+		}
+		setExpiries := tx.ReadWriteBucket(replaySetExpiryBucket)
+
+		var expiredSets [][]byte
+		err := setExpiries.ForEach(func(id, v []byte) error {
+			if numExpiredSets >= MaxNumberOfSetsRemovedPerBlock {
+				return breakForEach
```

Comment:
can `return nil`

Reply:
I see, I think then that the error catching should do:
`if err != nil && err != breakForEach` or use a switch statement

Reply:
I'd rather leave that to someone who knows `go` :) This is just a draft to get a discussion going.